aten/src/ATen/native/cuda/Unique.cu

```diff
@@ -86,6 +88,13 @@ std::tuple<Tensor, Tensor, Tensor> unique_cuda_template(
   auto options = self.options().dtype(kLong);
   Tensor output = self.clone(at::MemoryFormat::Contiguous).reshape(-1);
   int64_t num_inp = output.numel();
+  if (num_inp <= INT_MAX) {
+    // A common case is using CUB that requires fewer cuda stream synchronize
+    // but defined in a separate file because thrust and ATen/cuda/cub.cuh
+    // can't be used at the same compilation unit.
+    return internal::unique_cuda_template_cub<scalar_t>(
+        self, consecutive, return_inverse, return_counts);
+  }
```

Comment:
How about removing the thrust path completely for the non-dim unique? Because the thrust path does not support `num_inp > INT_MAX` as well.